src/apps/schema/src/app/components/SelectModelParentInput.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  Box,
+  InputLabel,
+  TextField,
+  Tooltip,
+  Autocomplete,
+} from "@mui/material";
+import { useMemo } from "react";
+import InfoRoundedIcon from "@mui/icons-material/InfoRounded";
+import { cloneDeep } from "lodash";
+import { useGetContentNavItemsQuery } from "../../../../../shell/services/instance";
+import { ContentNavItem, ModelType } from "../../../../../shell/services/types";
+
+type SelectModelParentInputProps = {
+  value: string;
+  onChange: (value: string) => void;
+  modelType: ModelType;
+};
+
+export const SelectModelParentInput = ({
+  value,
+  onChange,
+  modelType,
+}: SelectModelParentInputProps) => {
+  const { data: navItems } = useGetContentNavItemsQuery();
+
+  const parents = useMemo(() => {
+    if (navItems) {
+      const _navItems = cloneDeep(navItems);
```

Comment:
Why clone? This seems to only operate on the array and not object elements. Would creating a new array be more efficient?